src/services/uploadService.ts

```diff
@@ -0,0 +1,396 @@
+            canvas.toBlob(async (blob) => {
+                resolve(await this.getUint8ArrayView(blob));
+            })
+        });
+        return thumbnail;
+    }
+
+    private async getUint8ArrayView(file): Promise<Uint8Array> {
+        return await new Promise((resolve, reject) => {
+            const reader = new FileReader()
+
+            reader.onabort = () => reject('file reading was aborted')
+            reader.onerror = () => reject('file reading has failed')
+            reader.onload = () => {
+                // Do whatever you want with the file contents
+                const result = typeof reader.result === "string" ? new TextEncoder().encode(reader.result) : new Uint8Array(reader.result);
+                resolve(result);
+            }
+            reader.readAsArrayBuffer(file)
+        });
+    }
+
+    private async getUploadURL(token: string, count: number) {
+        if (this.uploadURLs.isEmpty()) {
+            await this.fetchUploadURLs(token, count);
+        }
+        return this.uploadURLs.pop();
+    }
+
+    private async fetchUploadURLs(token: string, count: number): Promise<void> {
+        if (!this.uploadURLFetchInProgress) {
+            this.uploadURLFetchInProgress = HTTPService.get(`${ENDPOINT}/files/upload-urls`,
+                {
+                    token: token,
+                    count: Math.min(50, count).toString()  //m4gic number
+                })
+            const response = await this.uploadURLFetchInProgress;
+
+            this.uploadURLFetchInProgress = null;
+            this.uploadURLs.push(response.data["urls"]);
+        }
+        return this.uploadURLFetchInProgress;
+    }
+
+    private async putFile(fileUploadURL: uploadURL, file: Uint8Array | string): Promise<string> {
+        const fileSize = file.length.toString();
+        await HTTPService.put(fileUploadURL.url, file, null, { contentLengthHeader: fileSize })
+        return fileUploadURL.objectKey;
+    }
+
+    private async getExifData(recievedFile) {
+        const exifData: any = await new Promise((resolve, reject) => {
+            const reader = new FileReader()
+            reader.onload = () => {
+                resolve(EXIF.readFromBinaryFile(reader.result));
+            }
+            reader.readAsArrayBuffer(recievedFile)
+        });
+        if (!exifData)
+            return null;
+        return {
+            location: this.getLocation(exifData),
+            creationTime: this.getUNIXTime(exifData)
+        };
+    }
+    private getUNIXTime(exifData: any) {
+        if (!exifData.DateTimeOriginal)
+            return null;
+        let dateString: string = exifData.DateTimeOriginal;
+        var parts = dateString.split(' ')[0].split(":");
+        var date = new Date(Number(parts[0]), Number(parts[1]) - 1, Number(parts[2]));
+        return date.getTime() * 1000;
+    }
+
+    private getLocation(exifData) {
+
+        if (!exifData.GPSLatitude)
+            return { lat: null, lon: null };
```

Comment:
[question]
@vishnukvmd  Should we also have a check for 0,0 coordinates?

Reply:
should I make latitude longitude fallback value could be made 0,0 coordinate?

Reply:
No, that looks like a [legit location](https://www.google.com/maps/place/0%C2%B000'00.0%22N+0%C2%B000'00.0%22E/@12.5735552,-4.8134094,3.49z/data=!4m2!3m1!1s0x0:0x0). Omit both `latitude` and `longitude` attributes.

Reply:
I meant that if the location present in the EXIF data is `0,0`, then most probably it's wrong. So, should we revert back to `null` in such cases?

Reply:
ohk....The exif data either has the correct location or is doesn't have location property at all. I haven't found any example when it was 0,0. None the less will add a check and revert back

Reply:
If the EXIF data suggests that it is `0,0`, let's accept it as is. Who knows where the user has been. 🤷